tests/client/test_auth.py

```diff
@@ -527,6 +527,47 @@ async def test_digest_auth_incorrect_credentials() -> None:
     assert len(response.history) == 1
 
 
+@pytest.mark.asyncio
+async def test_digest_auth_reuses_challenge() -> None:
+    url = "https://example.org/"
+    auth = DigestAuth(username="tomchristie", password="password123")
```

Comment:
Eeek... you've included my password in clear text. 😬
I'd suggest anonymising it like this instead...

```suggestion
    auth = DigestAuth(username="user", password="password123")
```